cpp/examples/luma.gl/animometer.cc

```diff
@@ -109,53 +109,51 @@ auto createSampleData(int count) -> std::vector<ShaderData> {
   return shaderData;
 }
 
-}  // anonymous namespace
+auto makeWebGPUTable(wgpu::Device device) -> std::shared_ptr<garrow::Table> {
+  auto positionsDescriptor = garrow::AttributeDescriptor{"positions", arrow::fixed_size_list(arrow::float32(), 4)};
+  auto colorsDescriptor = garrow::AttributeDescriptor{"colors", arrow::fixed_size_list(arrow::float32(), 4)};
 
-struct Attribute {
-  const char* name;
-  void* data;
-  size_t bytes;
-};
+  auto positionsField = std::make_shared<garrow::Field>("positions", positionsDescriptor);
+  auto colorsField = std::make_shared<garrow::Field>("colors", colorsDescriptor);
 
-auto makeWebGPUTable(wgpu::Device device, const std::vector<Attribute>& attributes) {
-  std::vector<wgpu::Buffer> buffers;
-  for (auto const attribute : attributes) {
-    auto buffer = utils::createBufferFromData(device, attribute.data, attribute.bytes, wgpu::BufferUsage::Vertex);
-    buffers.push_back(buffer);
-  }
-  return buffers;
+  std::vector<std::shared_ptr<garrow::Field>> fields{positionsField, colorsField};
+  auto schema = std::make_shared<garrow::Schema>(fields);
+
+  std::vector<Vector4<float>> positionData{
+      {0.0f, 0.1f, 0.0f, 1.0f}, {-0.1f, -0.1f, 0.0f, 1.0f}, {0.1f, -0.1f, 0.0f, 1.0f}};
+  auto positionsArray = std::make_shared<garrow::Array>(device, positionsDescriptor, positionData);
+
+  std::vector<Vector4<float>> colorData{{1.0f, 0.0f, 0.0f, 1.0f}, {0.0f, 1.0f, 0.0f, 1.0f}, {0.0f, 0.0f, 1.0f, 1.0f}};
+  auto colorsArray = std::make_shared<garrow::Array>(device, colorsDescriptor, colorData);
+
+  std::vector<std::shared_ptr<garrow::Array>> arrays{positionsArray, colorsArray};
+
+  return std::make_shared<garrow::Table>(schema, arrays);
 }
 
+}  // anonymous namespace
+
 int main(int argc, const char* argv[]) {
   GLFWAnimationLoop animationLoop{};
   wgpu::Device device = *(animationLoop.device.get());
 
-  lumagl::Model model{animationLoop.device, {vs, fs, {{}, {}}, {{sizeof(ShaderData)}}}};
+  auto attributes = makeWebGPUTable(device);
+  Model model{animationLoop.device, {vs, fs, attributes->schema(), {{sizeof(ShaderData)}}}};
```

Comment:
Note for Model API (not for this PR): We are going to need to be able to supply two schemas here. The primitive table schema, and the instance table schema.

This presumably also makes the location deduction trickier. I think we will eventually need to be able to list locations more explicitly, e.g. in the attribute descriptors.

There is also a third wrinkle. There can be an array of indices, that are not possible to represent in an arrow table. We need to allow the user to specify that this is an indexed model as well. It may not need to be done during creation. Maybe we have a `Model.setIndices` that trigger use of `drawIndexed` instead of `draw`.